lib/rubocop/cop/rspec/let_setup.rb

```diff
@@ -45,10 +51,19 @@ def on_block(node)
         private
 
         def unused_let_bang(node)
-          let_bang(node) do |method_send, method_name|
+          child_let_bang(node) do |method_send, method_name|
             yield(method_send) unless method_called?(node, method_name)
           end
         end
+
+        def child_let_bang(node)
+          lets = RuboCop::RSpec::ExampleGroup.new(node).lets
+
+          lets.each do |let|
+            node_with_name = let_bang(let)
+            yield node_with_name if node_with_name
+          end
+        end
```

Comment:
Just an option:
```ruby
        def child_let_bang(node, &block)
          RuboCop::RSpec::ExampleGroup.new(node)
            .lets
            .map { |let| let_bang(let) }
            .compact
            .each(&block)
        end
```
could be cleaner with `filter_map`, but it's Ruby 2.7 only AFAIR.

I think your current implementation is superior.